### PR TITLE
my hw jdbc

### PR DESCRIPTION
**Pull request by VolodymyrKachan**

### CODE REVIEW

File: src/main/java/mate/jdbc/dao/ManufacturerDao.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package mate.jdbc.dao;
+
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.model.Manufacturer;
+
+public interface ManufacturerDao {
+    Manufacturer create(Manufacturer manufacturer);
```
Comment (JustElvis) [inline src/main/java/mate/jdbc/dao/ManufacturerDao.java:8]: create method must separated from previous line

**Review by JustElvis — CHANGES_REQUESTED** [review]

**Review by JustElvis — APPROVED** [review]

File: src/main/java/mate/jdbc/dao/ManufacturerDao.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package mate.jdbc.dao;
+
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.model.Manufacturer;
+
+public interface ManufacturerDao {
+
```
Comment (Noct2000) [inline src/main/java/mate/jdbc/dao/ManufacturerDao.java:8]: delete empty line 
```suggestion
```

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+
```
Comment (Noct2000) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:18]:
```suggestion
```

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+                + " values(?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createManufacturerStatement = connection
+                        .prepareStatement(insertManufacturerRequest,
+                                Statement.RETURN_GENERATED_KEYS)) {
+            createManufacturerStatement.setString(1, manufacturer.getName());
+            createManufacturerStatement.setString(2, manufacturer.getCountry());
+            createManufacturerStatement.executeUpdate();
+            ResultSet keysGenerate = createManufacturerStatement.getGeneratedKeys();
+            if (keysGenerate.next()) {
```
Comment (Noct2000) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:34]: this check is redundant

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement getManufacturerStatement = connection
+                        .prepareStatement(getManufacturer)) {
+            getManufacturerStatement.setLong(1, id);
+            ResultSet resultSet = getManufacturerStatement.executeQuery();
+            Manufacturer manufacturer = null;
+            if (resultSet.next()) {
+                manufacturer = new Manufacturer(id,
+                        resultSet.getString("name"),
+                        resultSet.getString("country"));
```
Comment (Noct2000) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:79]: move logic of parsing `resultSet` to private method and replace this logic with new private method

**Review by Noct2000 — CHANGES_REQUESTED** [review]

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+                + " values(?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createManufacturerStatement = connection
+                        .prepareStatement(insertManufacturerRequest,
+                                Statement.RETURN_GENERATED_KEYS)) {
+            createManufacturerStatement.setString(1, manufacturer.getName());
+            createManufacturerStatement.setString(2, manufacturer.getCountry());
+            createManufacturerStatement.executeUpdate();
+            ResultSet keysGenerate = createManufacturerStatement.getGeneratedKeys();
+            while (keysGenerate.next()) {
```
Comment (Noct2000) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:33]:
```suggestion
            if (keysGenerate.next()) {
```

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement getAllManufacturersStatement = connection
+                         .prepareStatement(selectQuery)) {
+            ResultSet resultSet = getAllManufacturersStatement.executeQuery();
+            while (resultSet.next()) {
+                Manufacturer manufacturer = new Manufacturer(resultSet
+                        .getObject("id", Long.class),
+                        resultSet.getString("name"),
+                        resultSet.getString("country"));
+                allManufacturers.add(manufacturer);
```
Comment (Noct2000) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:93]: replace it with private method

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+                manufacturer = new Manufacturer(id,
+                        resultSet.getString("name"),
+                        resultSet.getString("country"));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get manufacturer by id = " + id, e);
+        }
+        return Optional.ofNullable(manufacturer);
+    }
+}
```
Comment (Noct2000) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:127]:
```suggestion
    private Manufacturer parseManufacturer(ResultSet resultSet, Long id) // or getManufacturer
    {
        // here you need to create a new manufacturer and set fields from resultSet, you need catch exception in the method where you call this private method 
}
```

**Review by Noct2000 — CHANGES_REQUESTED** [review]

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+            deleteManufacturerStatement.setLong(1, id);
+            return deleteManufacturerStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete manufacturer by id = " + id, e);
+        }
+    }
+
+    private static Manufacturer checkResulSet(ResultSet resultSet, Long id) throws SQLException {
+        Manufacturer manufacturer = null;
+        if (resultSet.next()) {
```
Comment (Noct2000) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:112]: move this check to the methods where are you calling this private method

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+                 PreparedStatement deleteManufacturerStatement = connection
+                         .prepareStatement(deleteQuery)) {
+            deleteManufacturerStatement.setLong(1, id);
+            return deleteManufacturerStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete manufacturer by id = " + id, e);
+        }
+    }
+
+    private static Manufacturer checkResulSet(ResultSet resultSet, Long id) throws SQLException {
```
Comment (Noct2000) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:110]:
```suggestion
    private Manufacturer parseManufacturer(ResultSet resultSet, Long id) throws SQLException {
```

**Review by Noct2000 — CHANGES_REQUESTED** [review]

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+                 PreparedStatement deleteManufacturerStatement = connection
+                         .prepareStatement(deleteQuery)) {
+            deleteManufacturerStatement.setLong(1, id);
+            return deleteManufacturerStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete manufacturer by id = " + id, e);
+        }
+    }
+
+    private static Manufacturer parseManufacturer(ResultSet resultSet, Long id)
```
Comment (Noct2000) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:115]: and you can get id from `resultSet`
```suggestion
    private Manufacturer parseManufacturer(ResultSet resultSet, Long id)
```

**Review by Noct2000 — APPROVED** [review]

**Pull request closed without merging**